packages/kbn-apm-synthtrace-client/src/lib/timerange.ts

```diff
@@ -15,8 +15,20 @@ export class Timerange {
     return new Interval({ from: this.from, to: this.to, interval });
   }
 
-  ratePerMinute(rate: number) {
-    return this.interval(`1m`).rate(rate);
+  ratePerMinute(ratePerMin: number) {
+    const intervalPerSecond = Math.max(1, 60 / ratePerMin);
+
+    // rate per second
+    let interval = `${intervalPerSecond}s`;
+    let rate = (ratePerMin / 60) * intervalPerSecond;
+
+    // rate per minute
+    if (!Number.isInteger(rate) || !Number.isInteger(intervalPerSecond)) {
+      interval = '1m';
+      rate = rate * 60;
+    }
+
+    return this.interval(interval).rate(rate);
   }
```

Comment:
What is the value over interval(x).rate('1s')? I'm not a fan of it switching back to per minute if it's not a multiple of 60 either. I'd like to avoid too much magic, especially if it's not a clear improvement over the original API.

Reply:
The problem is that we can't specify the rate in tpm without seeing spikes. This solves that.

Reply:
> What is the value over interval(x).rate('1s')? 

One of Synthtrace' strengths is that it creates predictable data where it's easy to see the relationship between the synthtrace scenarios, and how it shows up in the UI.

If you see 30tpm in the ui, you should be able to find those 30tpm in the scenario. But mostly we specify in tps in scenario to avoid spikes. 

>  I'm not a fan of it switching back to per minute if it's not a multiple of 60 either. I'd like to avoid too much magic, especially if it's not a clear improvement over the original API.

We could make this less magic by requiring rates that can be converted to tps evenly.


```
  ratePerMinute(ratePerMin: number) {
    const intervalPerSecond = Math.max(1, 60 / ratePerMin);

    // rate per second
    const interval = `${intervalPerSecond}s`;
    const rate = (ratePerMin / 60) * intervalPerSecond;

    // invalid rate
    if (!Number.isInteger(rate) || !Number.isInteger(intervalPerSecond)) {
      throw new Error(`${ratePerMin}tpm is not a valid rate.`)
    }

    return this.interval(interval).rate(rate);
  }